### ISSUE TITLE
Support environment variables for provider configuration

### ISSUE THREAD

**Issue by JohannesRudolph** [meshcloud/terraform-provider-meshstack#28]
this allows setting the credentials form an external source e.g. in CI/CD

**Comment by michael-follmann** [comment]
Of course, you can work around it and create variables in your Terraform root module(s) and provide these with environment variables (`TF_VAR_meshstack_api_key` and `TF_VAR_meshstack_api_secret` in this case):

```terraform
variable "meshstack_api_key" {
  description = "API Key to authenticate against the meshStack API"
  type        = string
  sensitive   = true
}

variable "meshstack_api_secret" {
  description = "API Secret to authenticate against the meshStack API"
  type        = string
  sensitive   = true
}

provider "meshstack" {
  ...
  apikey    = var.meshstack_api_key
  apisecret = var.meshstack_api_secret
}
```

However, I'd rather just provide environment variables directly (e.g. `MESHSTACK_API_KEY` and `MESHSTACK_API_SECRET`) so that I don't have to declare additional variables.

Other Terraform providers support similar environment variables, for example:

- [AWS (e.g. `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`)](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#environment-variables)
- [Datadog (e.g. `DD_API_KEY`)](https://registry.terraform.io/providers/DataDog/datadog/latest/docs)
